### PR TITLE
Add option to request extra boot disk space for Batch Tasks

### PR DESCRIPTION
**Pull request by lathanh**

### CODE REVIEW

File: buildstockbatch/schemas/v0.3.yaml [hunk 1]
```diff
@@ -44,6 +44,7 @@ gcp-ar-spec:
 gcp-job-environment-spec:
   vcpus: int(min=1, max=224, required=False)
   memory_mib: int(min=512, required=False)
+  boot_disk_mib: int(required=False)
```
Comment (mfathollahzadeh) [inline buildstockbatch/schemas/v0.3.yaml:47]: Can you add the default as a comment? also is this extra boot disk?

File: docs/project_defn.rst [hunk 2]
```diff
@@ -298,6 +298,11 @@ using `GCP Batch <https://cloud.google.com/batch>`_ and `Cloud Run <https://clou
     *  ``vcpus``: Number of CPUs to allocate for running each simulation. Default: 1.
     *  ``memory_mib``: Amount of RAM memory needed for each simulation in MiB. Default: 1024.
        For large multifamily buildings this works better if set to 2048.
+    *  ``boot_disk_mib``: Optional. Extra boot disk size in MiB for each task. This affects how
+       large the boot disk will be (see the `Batch OS environment docs`_ for details) of the
+       machine(s) running simulations (which is the disk used by simulations). This will likely need
+       to be set to at least 2,048 if more than 8 simulations will be run in parallel on the same
+       machine (i.e., when vCPUs per machine_type ÷ vCPUs per sim > 8). Default: None.
```
Comment (mfathollahzadeh) [inline docs/project_defn.rst]: I thought we were discussing the default being 30 GB? Do you mean the default extra?

**Review by mfathollahzadeh — COMMENTED** [review]
Thanks, Robert! Just added some questions for clarity

File: buildstockbatch/schemas/v0.3.yaml (hunk 1 above)
Comment (lathanh) [reply buildstockbatch/schemas/v0.3.yaml:47]: > Can you add the default as a comment?

We're not generally documenting parameters here (but in the `docs/project_defn.rst` documentation file).

> also is this extra boot disk?

This _may_ add extra size the to _the_ boot disk (it doesn't add an extra disk), more on this in [the other thread](https://github.com/rewiringamerica/buildstockbatch/pull/57#discussion_r1447947833).

File: docs/project_defn.rst (hunk 2 above)
Comment (lathanh) [reply docs/project_defn.rst]: There are two levels of default:

1. The default value here, `None`, which means "Don't make any explicit request to Batch for disk size." By not making such an explicit request...
2. When Batch calculates what the disk size will be, that will end up being 30 GB — at the time of writing!

I want to leverage their default behavior; i.e., I don't want to introduce a redundant default. I also don't want to introduce code or documentation that would be inaccurate if they change their behavior (e.g., if they changed the minimum from 30 GB to 40 GB).

That said, I've added a bit more (in parenthesis) to the documentation about the default.

**Review by mfathollahzadeh — APPROVED** [review]

**Review by nweires — APPROVED** [review]

**Pull request merged**